PermissionsScraper/PermissionsScraper/Helpers/ProtectedApiCallHelper.cs

```diff
@@ -0,0 +1,55 @@
+﻿// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace PermissionsScraper.Helpers
+{
+    /// <summary>
+    /// Helper class to call a protected API and returns the result of the call
+    /// </summary>
+    internal static class ProtectedApiCallHelper
+    {
+        /// <summary>
+        /// Calls a protected Web API
+        /// </summary>
+        /// <param name="webApiUrl">Url of the Web API to call</param>
+        /// <param name="accessToken">Access token used as a bearer security token to call the Web API</param>
+        /// <returns>The response of the call to the Web API</returns>
+        internal static async Task<string> CallWebApiAsync(string webApiUrl, string accessToken)
+        {
+            if (string.IsNullOrEmpty(webApiUrl))
+            {
+                throw new ArgumentNullException(nameof(webApiUrl), "Parameter cannot be null or empty");
+            }
+            if (string.IsNullOrEmpty(accessToken))
+            {
+                throw new ArgumentNullException(nameof(accessToken), "Parameter cannot be null or empty");
+            }
+
+            var httpClient = new HttpClient();
+            var defaultRequestHeaders = httpClient.DefaultRequestHeaders;
+            if (defaultRequestHeaders.Accept == null || !defaultRequestHeaders.Accept.Any(m => m.MediaType == "application/json"))
+            {
+                httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
```

Comment:
based on the condition above `httpClient.DefaultRequestHeaders.Accept` can be `null` here.

Reply:
How do you mean?
We are checking if it is null, then setting it to `application/json`

Reply:
If `httpClient.DefaultRequestHeaders.Accept` is `null`, how are you allowed to call `Add` on it?

Reply:
Yea, it will never be null, so no need to check for it.